openvdb/points/PointMask.h

```diff
@@ -240,41 +237,38 @@ inline typename GridT::Ptr convertPointsToScalar(
 
     if (points.constTree().leafCount() == 0)            return grid;
 
-    const bool useGroup = !includeGroups.empty() || !excludeGroups.empty();
+    FilterT newFilter(filter);
 
     // early exit if mask and no group logic
 
-    if (std::is_same<ValueT, bool>::value && !useGroup) return grid;
+    if (std::is_same<ValueT, bool>::value && filter.state() == index::ALL) return grid;
 
     // evaluate point group filters to produce a subset of the generated mask
 
     tree::LeafManager<GridTreeT> leafManager(*tree);
 
-    if (useGroup) {
-        // build mask from points in parallel only where filter evaluates to true
-        const auto leaf = points.constTree().cbeginLeaf();
-        MultiGroupFilter filter(includeGroups, excludeGroups, leaf->attributeSet());
-        PointsToScalarOp<GridT, PointDataGridT, MultiGroupFilter> pointsToScalarOp(
-            points, filter);
-        tbb::parallel_for(leafManager.leafRange(), pointsToScalarOp);
-    }
-    else {
-        NullFilter filter;
+    if (filter.state() == index::ALL) {
+        NullFilter nullFilter;
         PointsToScalarOp<GridT, PointDataGridT, NullFilter> pointsToScalarOp(
-            points, filter);
+            points, nullFilter);
+        tbb::parallel_for(leafManager.leafRange(), pointsToScalarOp);
+    } else {
+        // build mask from points in parallel only where filter evaluates to true
+        FilterT newFilter(filter);
+        PointsToScalarOp<GridT, PointDataGridT, FilterT> pointsToScalarOp(
+            points, newFilter);
```

Comment:
threaded argument with foreach?

Reply:
Again this is part of the changes proposed in https://github.com/dreamworksanimation/openvdb/pull/232. If we can get this PR accepted first, I'll then make the relevant changes to update that one.